client/index.js

```diff
@@ -19,31 +19,47 @@ if (!window.Promise) {
 
 const {
   __NEXT_DATA__: {
-    component,
-    errorComponent,
     props,
     err,
     pathname,
-    query
+    query,
+    buildId
   },
   location
 } = window
 
-const Component = evalScript(component).default
-const ErrorComponent = evalScript(errorComponent).default
-let lastAppProps
-
-export const router = createRouter(pathname, query, getURL(), {
-  Component,
-  ErrorComponent,
-  err
-})
+const pageLoader = window.__NEXT_PAGE_LOADER__ = new PageLoader(buildId)
+if (window.__NEXT_LOADED_PAGES__) {
+  window.__NEXT_LOADED_PAGES__.forEach((fn) => fn())
+  delete window.__NEXT_LOADED_PAGES__
```

Comment:
Actually, delete is not bad. You need to use it wisely. Delete is bad when used inside a constructor. Because that's de-optimization. That's also valid for when we are creating a lot of objects from that class. 

We need to use it wisely. And we need to delete when needed. Anyway, it's not needed here and having it also not a problem :)